sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -192,3 +260,35 @@ def transform(self, X):
         _map_to_bins(X, self.bin_thresholds_, self.missing_values_bin_idx_,
                      binned)
         return binned
+
+    def make_known_categories_bitsets(self):
+        """Create bitsets of known categories.
+
+        Returns
+        -------
+        - known_cat_bitsets : ndarray of shape (n_categorical_features, 8)
+            Array of bitsets of known categories, for each categorical feature.
+        - f_idx_map : ndarray of shape (n_features,)
+            Map from original feature index to the corresponding index in the
+            known_cat_bitsets array.
+        """
+
+        categorical_features_indices = np.flatnonzero(self.is_categorical_)
```

Comment:
Why `np.flatnonzero` instead of just `np.zero`?

Reply:
Because `np.zero` doesn't exist ;p
If you meant `np.nonzero`, it returns a tuple instead of just an array